### ISSUE TITLE
errors in lab three

### ISSUE THREAD

**Issue by Zewen-Yang** [eje24/iap-diffusion-labs#8]
1. The sample function should return "self.std  * torch.randn(num_samples, *self.shape).to(self.dummy.device), None"

class IsotropicGaussian(nn.Module, Sampleable):
    ...
    def sample(self, num_samples) -> Tuple[torch.Tensor, torch.Tensor]:
        return self.std ** 2 * torch.randn(num_samples, *self.shape).to(self.dummy.device), None

2. The sample function should return "self.p_data.sample(num_samples)"

class GaussianConditionalProbabilityPath(ConditionalProbabilityPath):
    ...
    def sample_conditioning_variable(self, num_samples: int) -> torch.Tensor:
        """
        Samples the conditioning variable z and label y
        Args:
            - num_samples: the number of samples
        Returns:
            - z: (num_samples, c, h, w)
            - y: (num_samples, label_dim)
        """
        return p_data.sample(num_samples)

**Comment by eje24** [comment]
Great catches, @Zewen-Yang ! That first one is actually quite embarrassing, oops!  😆
